### DISCUSSION TITLE
How to properly use enable-glob when disabling everything else

### DISCUSSION THREAD

**Discussion by istudyatuni** [rustic-rs/rustic#700] [Q&A]
I want to use glob not for "blacklisting" files/folders, but for "whitelisting", so I do

```toml
[[backup.sources]]
source = "/tmp/rustic-test/dir"
glob = [
    "!*",
    "/tmp/rustic-test/dir/a",
    "/tmp/rustic-test/dir/b/b/a.txt",
]
```

but in this case folder's content and file `b/b/a.txt` not included in backup, so I should add more globs

```toml
[[backup.sources]]
source = "/tmp/rustic-test/dir"
glob = [
    "!*",

    # for adding folder and its content I should specify both
    "/tmp/rustic-test/dir/a",
    "/tmp/rustic-test/dir/a/**/*",

    # for adding one file in nested folders I should specify all folders
    "/tmp/rustic-test/dir/b",
    "/tmp/rustic-test/dir/b/b",
    "/tmp/rustic-test/dir/b/b/a.txt",
]
```

Is it possible to specify only folder and file?

Also worth noting that "blacklisting back" like

```toml
[[backup.sources]]
source = "/tmp/rustic-test/dir"
glob = [
    "!*",
    "/tmp/rustic-test/dir/a",
    "!/tmp/rustic-test/dir/a/b.txt",  # <- this is not work
    "/tmp/rustic-test/dir/a/**/*",
    "!/tmp/rustic-test/dir/a/b.txt",  # <- this works
]
```

works only if specify anti-pattern after allowing everything in directory

**Comment by aawsome** [comment]
There have been a few open issues about including/excluding. And I agree that it is not perfectly usable the way it currently is. I'll open a general include/exclude issue where we can discuss how to best rework that whole part.
